pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -123,32 +123,61 @@ func getSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.Repli
 		}
 	}
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
+	rcPod, rsPod, stsPod := false, false, false
+	for _, ref := range pod.OwnerReferences {
+		if ref.Controller != nil && *ref.Controller == true {
+			switch ref.Kind {
+			case "ReplicationController":
+				if ref.APIVersion == "v1" {
```

Comment:
can you also verify what Kind/APIVersion was set by each of these workload controllers on pods in 1.14?

Reply:
I think it is, as I checked out code to v1.14.0 and grep `staging/src/k8s.io/api` directory, `type ReplicationController struct` only exits in core/v1/types.go file (I'm not sure it is accurate). The same as other workloads.